### PR TITLE
Temp auth

### PR DESCRIPTION
**Pull request by Kothalkarpriya**
- added login and signup page
- added responsiveness for every screen sizes

### CODE REVIEW

File: src/context/AuthContext.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+  };
+
+  const signup = async (user) => {
+    try {
+      const data = await axios.post(`/user/signup`, user);
+      localStorage.setItem("user", JSON.stringify(data));
+      authDispatch({ type: "SIGNUP", payload: data });
+      navigate(from, { replace: true });
+    } catch (error) {}
```
Comment (rkapoor10) [inline src/context/AuthContext.jsx]: give something after catching error

File: src/pages/Auth/SignUp.jsx [hunk 2]
```diff
@@ -1,5 +1,109 @@
...
+            <div className="input-block">
+              <label for="#input-name" className="text-align-left">
+                First Name
+              </label>
+              <input
+                type="text"
+                id="input-name"
+                placeholder="First Name"
+                required
+                // onChange={(e) => changeInputHandler(e)}
```
Comment (rkapoor10) [inline src/pages/Auth/SignUp.jsx:28]: why commented?

File: src/pages/Auth/SignUp.jsx [hunk 3]
```diff
@@ -1,5 +1,109 @@
...
+            <div className="input-block">
+              <label for="#input-lastname" className="text-align-left">
+                Last Name
+              </label>
+              <input
+                type="text"
+                id="input-lastname"
+                placeholder="Last Name"
+                required
+                // onChange={(e) => changeInputHandler(e)}
```
Comment (rkapoor10) [inline src/pages/Auth/SignUp.jsx:41]: here also

File: src/pages/Auth/SignUp.jsx [hunk 4]
```diff
@@ -1,5 +1,109 @@
...
+          <div className="input-block">
+            <label for="#input-email" className="text-align-left">
+              Email Address
+            </label>
+            <input
+              type="email"
+              id="input-email"
+              placeholder="username@email.com"
+              required
+              //   onChange={(e) => changeInputHandler(e)}
```
Comment (rkapoor10) [inline src/pages/Auth/SignUp.jsx:55]: same as above

File: src/pages/Auth/SignUp.jsx [hunk 5]
```diff
@@ -1,5 +1,109 @@
...
+            <label for="input-pass" className="text-align-left">
+              Password
+            </label>
+            <input
+              type="password"
+              id="input-pass"
+              className="input-password"
+              placeholder="*************"
+              required
+              //   onChange={(e) => changeInputHandler(e)}
```
Comment (rkapoor10) [inline src/pages/Auth/SignUp.jsx:69]: same as above. If you don't need them; add a comment.

File: src/route/RoutePath.jsx [hunk 6]
```diff
@@ -2,16 +2,38 @@ import { Routes, Route } from "react-router-dom";
...
+          <AuthRoute>
+            <Quiz />
+          </AuthRoute>
+        }
+      />
+      <Route
+        path="/rules"
+        element={
+          <AuthRoute>
+            <Rules />{" "}
```
Comment (rkapoor10) [inline src/route/RoutePath.jsx]: `{" "}` remove this

**Review by rkapoor10 — APPROVED** [review]
The code looks good to merge. Go through the comments.

File: src/context/AuthContext.jsx (hunk 1 above)
Comment (Kothalkarpriya) [reply src/context/AuthContext.jsx]: Yes Added

File: src/pages/Auth/SignUp.jsx (hunk 3 above)
Comment (Kothalkarpriya) [reply src/pages/Auth/SignUp.jsx:41]: Ok added

File: src/pages/Auth/SignUp.jsx (hunk 5 above)
Comment (Kothalkarpriya) [reply src/pages/Auth/SignUp.jsx:69]: Ok added

File: src/route/RoutePath.jsx (hunk 6 above)
Comment (Kothalkarpriya) [reply src/route/RoutePath.jsx]: Yes Removed

File: src/pages/Auth/SignUp.jsx (hunk 4 above)
Comment (Kothalkarpriya) [reply src/pages/Auth/SignUp.jsx:55]: Yes added comment

File: src/pages/Auth/SignUp.jsx (hunk 2 above)
Comment (Kothalkarpriya) [reply src/pages/Auth/SignUp.jsx:28]: This is needed for the further feature

**Pull request merged**